### ISSUE TITLE
Add integration test suite for CORS requests

### ISSUE THREAD

**Issue by tobiasdiez** [apollographql/apollo-server#7044]
The [docs](https://www.apollographql.com/docs/apollo-server/security/cors#configuring-cors-options-for-apollo-server) mention that in v4 the user is responsible for setting up the CORS handler (and not the integration package). Nonetheless, it would be nice if there were integration tests that make sure that the recommended way is indeed working as intended.

For example, while developing https://github.com/apollo-server-integrations/apollo-server-integration-h3 I noticed that preflight requests shouldn't hit the apollo http handler (since it only accepts GET and POST). Thus, depending on the framework either the cors handler or the integration package should handle preflights. 

https://github.com/JabRef/JabRefOnline/blob/214c8b991827a2b71f6c0dd675a12280e8b874a9/server/e2e.test.ts#L44-L78 can serve as a very simple starting point. The usage could look as follows:
```ts
defineIntegrationTestSuite(
  async (
    serverOptions: ApolloServerOptions<BaseContext>,
    testOptions?: CreateServerForIntegrationTestsOptions
  ) => {

    // the usual stuff, setting up the apollo server as well as the app

    if (testOptions.corsOptions) {
       // add cors handler (depends on the framework of course), something like
       app.use('/', cors(testOptions.corsOptions))
    }
}
```

**Comment by glasser** [comment]
That's an interesting point, though to some degree this is going to be testing the setup code you use when invoking the integration test suite more than the actual integration itself.

I think we could just write those tests now, though — after all, the built-in test suite invocations do already use the Express `cors` middleware with the default options. So adding some tests that `OPTIONS` works, etc, probably could just work without needing any new test options?

**Comment by tobiasdiez** [comment]
> to some degree this is going to be testing the setup code you use when invoking the integration test suite more than the actual integration itself.

I agree completely! But the code that you use to setup the integration should be (more or less) what you recommended users of the integration to use in their apps. So yes, you essentially test your documentation.

> So adding some tests that OPTIONS works, etc, probably could just work without needing any new test options?

Good point. I guess it depends on how throughout you want to be, e.g. test what happens if the requests's origin is not accepted. But I guess the default cors options serve already as a pretty good starting point.

I also remember that the diagnostics tool for apollo studio had some tests surrounding cors, so maybe these can be easily reused?

**Comment by glasser** [comment]
Out of curiosity, are you interested in adding these tests?

**Comment by tobiasdiez** [comment]
Interested yes, but I fear I don't have the time for this right now. (the h3 integration already took me relatively far away from what I actually have to do)
